src/main/java/Duke.java

```diff
@@ -1,10 +1,176 @@
+                        try {
+                            st.nextToken();
+                            int endIdx = curr.indexOf(" /by ");
+                            if (endIdx == -1) {
+                                new DukeException(out, "OOPS!!! The deadline cannot be empty.");
+                            } else {
+                                currTask = new Deadline(curr.substring(8, endIdx), " (by:" + curr.substring(endIdx + 4) + ")");
+                                tasks.add(currTask);
+                                currTask.addTask(out, tasks.size());
+                            }
+                        } catch (NoSuchElementException e1) {
+                            new DukeException(out, "OOPS!!! The description of a deadline cannot be empty.");
+                        }
+                        break;
+                    case event:
+                        try {
+                            st.nextToken();
+                            int endIdx = curr.indexOf(" /at ");
+                            if (endIdx == -1) {
+                                new DukeException(out, "OOPS!!! The venue cannot be empty.");
+                            } else {
+                                currTask = new Deadline(curr.substring(8, endIdx), " (by:" + curr.substring(endIdx + 4) + ")");
+                                tasks.add(currTask);
+                                currTask.addTask(out, tasks.size());
+                            }
+                        } catch (NoSuchElementException e1) {
+                            new DukeException(out, "OOPS!!! The description of an event cannot be empty.");
+                        }
+                        break;
+                }
+            } catch (Exception e) {
+                new DukeException(out, "OOPS!!! I'm sorry, but I don't know what that means :-(");
+            }
+            out.flush();
+            curr = br.readLine();
+            st = new StringTokenizer(curr);
+        }
     }
 }
+
+class Task{
+    public String type = "[T]";
+    public String status = "[" + "\u2718" +"]" ;
+    public String name;
+    public Task(String x){
+        this.name = x;
+    }
+    public void print(BufferedWriter out) throws Exception{
+        out.write(this.type + this.status + this.name + "\n");
+    }
+    public void done(BufferedWriter out) throws Exception{
+        this.status ="[" + "\u2713" +"]" ;
+        out.write("Nice! I've marked this task as done:\n");
+        this.print(out);
+    }
+    public void addTask(BufferedWriter out, int number) throws Exception {
+        out.write("Got it. I've added this task:\n");
+        this.print(out);
+        if(number > 1) {
+            out.write("Now you have " + String.valueOf(number) + " tasks in the list.\n");
+        }else{
+            out.write("Now you have 1 task in the list.\n");
+        }
+    }
+}
+class Deadline extends Task{
+    public String ddl;
+    public Deadline (String x, String d){
```

Comment:
It will be good if you can rename the variables x and d to something more meaningful, to improve the readability of your code.